### ISSUE TITLE
Preamble ToC to use anchors for level 2 and 3

### ISSUE THREAD

**Issue by cmc333333** [eregs/notice-and-comment#109]
Currently, we only display two levels of the preamble ToC. Further, the second level links to display _only_ the text in that subsection, rather than linking to an anchor within the larger, level one text.

* [ ] Add level 3
* [ ] Link to anchors within a section for level 2 and level 3

**Comment by jmcarp** [comment]
I can take a look at this. Like I was asking in #113, though, I'm wondering if we want to build the preamble toc someplace other than the templates first. What do you think @cmc333333?

**Comment by cmc333333** [comment]
Yes, I'd strongly recommend not building the ToC in the template

**Comment by jmcarp** [comment]
Right, I only mention it because that's where it lives now.

**Comment by cmc333333** [comment]
Probably related to #105

**Comment by jmcarp** [comment]
Before I check in another false start, how far off is something like the following?

```python
PreambleSect = namedtuple('PreambleSect', ['depth', 'title', 'url', 'children'])

def make_preamble_toc(nodes, depth=1):
	if depth >= 3:
		return None
	return [
		PreambleSect(
			depth=depth,
			title=node['title'],
			url=reverse(
				'chrome_preamble',
				kwargs={'paragraphs': '/'.join(node['label'])},
			),
			children=make_preamble_toc(node.get('children', []), depth=depth + 1)
		)
		for node in nodes
		if node.get('title')
	]
```

Are we opposed to the templates inspecting `depth` and deciding how to render? Other things I'm missing? @cmc333333

**Comment by cmc333333** [comment]
Mostly on the mark. A few things:

We want the `url` to only be one level deep, e.g.
```
111_22-I -> 111_22/I#111_22-I
111_22-I-B -> 111_22/I#111_22-I-B
111_22-I-B-4 -> 111_22/I#111_22-I-B-4
```

The "depth" here can also be determined via the label length, if you don't want to carry it around. I don't have a strong preference on how much to inspect it in the template for the preamble. The CFR changes are a bit funkier with regards to nesting, so we'll probably want a "type" or signifier like "is_subpart" there, if you want to keep parity.

Something that might be worth considering: currently the rendered ToC flattens the nesting (this plays better with existing CSS/JS). If we want to be more semantic (e.g. nesting `ol`s), there might be some fallout with frontend land.

**Comment by cmc333333** [comment]
Resolved by eregs/regulations-site#217
